### ISSUE TITLE
Selected value for discriminator field doesn't make sense if response is a subschema

### ISSUE THREAD

**Issue by lei-sumo** [Rebilly/ReDoc#573]
For the `PetStore` example, if I add an endpoint to return a `Dog` by Id and reference `Pet`'s subschema `Dog`, I would expect the generated doc has `Dog` for the response schema. However, it is still showing `Pet` as the response type and default the selection to `Cat`. This is very confusing. As we already know its strong type, shouldn't it point to the correct value (and disable the dropbox ideally)?

Endpoint added:
```
'/dog/{dogId}':
    get:
      tags:
        - pet
      summary: Find dog by ID
      description: Returns a single dog
      operationId: getDogById
      parameters:
        - name: dogId
          in: path
          description: ID of dog to return
          required: true
          deprecated: true
          schema:
            type: integer
            format: int64
      responses:
        '200':
          description: successful operation
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Dog'
            application/xml:
              schema:
                $ref: '#/components/schemas/Dog'
        '400':
          description: Invalid ID supplied
        '404':
          description: Dog not found
      security:
        - api_key: []
```

Generated UI
![image](https://user-images.githubusercontent.com/15622224/43110155-39c78ade-8e9f-11e8-8511-81a9689a6362.png)

Expected UI
![image](https://user-images.githubusercontent.com/15622224/43110235-a8361c38-8e9f-11e8-9035-54ae1ee2329d.png)

I'm not too worried about it if only discriminator field is populated wrongly. But it actually renders other `Cat` specific fields and ignores `Dog` specific fields. This makes it unacceptable to release the documentation to our customers.

**Comment by RomanGotsiy** [comment]
I will investigate this.
Thanks for the detailed issue report

**Comment by lei-sumo** [comment]
Hi @RomanGotsiy, do you have any idea how to fix this?

**Comment by RomanGotsiy** [comment]
Yes, I started working on this but don't have enough time to finish it.
It's a big change so have to be very careful.

**Comment by lei-sumo** [comment]
Great. Thanks for the update.

**Comment by crussell52** [comment]
Hi -- new user of ReDoc... loving it... but I immediately ran into this.  Is there an update on progress?

**Comment by fabianvo** [comment]
I ran into the same issue and tried to fix it. Maybe my hack can help someone to come up with a better solution. Or at least start a discussion about improved discriminator support in ReDoc.

**Comment by vadzim-marchanka** [comment]
Hello. I am a new user of ReDoc. UI is awesome but the issue isn't fixed.
